File: java/parser/src/main/java/gov/lbl/parser/parser/Import_listVisitor.java
package gov.lbl.parser.parser;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Import_list;

import static java.util.stream.Collectors.toList;

import java.util.List;

public class Import_listVisitor extends modelicaBaseVisitor<Import_list> {
    @Override
    public Import_list visitImport_list(modelicaParser.Import_listContext ctx) {
        List<String> identifier_list = ctx.IDENT() == null ? null : ctx.IDENT().stream().map(identifier -> identifier.getText()).collect(toList());;
        return new Import_list(identifier_list);
    }
}

File: jsParser/parser/Import_listVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const Import_list = require('../domain/Import_list')

class Import_listVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitImport_list (ctx) {
    const identifier_list = []

    if (ctx.IDENT()) {
      ctx.IDENT().forEach(identifier => {
        identifier_list.push(identifier)
      })
    }

    return new Import_list.Import_list(identifier_list)
  }
};

Import_listVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitImport_list = this.visitImport_list
exports.Import_listVisitor = Import_listVisitor
